### ISSUE TITLE
Scanner Aborts/Crashes on Character Scan

### ISSUE THREAD

**Issue by Gilz96** [Andrewthe13th/Inventory_Kamera#113]
Hi my scan keeps crashing when scanning my characters. It stops when the scanner moves on to the second character. 
This error message was in the log:

Object reference not set to an instance of an object.
   at InventoryKamera.CharacterScraper.ScanCharacters() in B:\Projects\VisualStudio\InventoryKamera\InventoryKamera\InventoryKamera\CharacterScraper.cs:line 23
   at InventoryKamera.InventoryKamera.GatherData(Boolean[] formats, Boolean[] checkbox) in B:\Projects\VisualStudio\InventoryKamera\InventoryKamera\InventoryKamera\InventoryKamera.cs:line 94
   at InventoryKamera.Form1.<>c__DisplayClass21_0.<StartButton_Clicked>b__0() in B:\Projects\VisualStudio\InventoryKamera\InventoryKamera\InventoryKamera\Form1.cs:line 207

**Comment by Cupcak3** [comment]
What is the character that it crashes on?

**Comment by Gilz96** [comment]
It has crashed on Hu Tao, Tartaglia and Xingqui, I thought it might have been character based so I changed my party.

**Comment by Gilz96** [comment]
I just ran a character only scan and it worked perfectly

**Comment by Cupcak3** [comment]
Were the passing and failed scans done on different resolutions? It would be helpful to know if you changed anything between the fail and success.

**Comment by Gilz96** [comment]
All settings were the same. I only unticked the Weapons and Artifacts box in the scan settings.

I have since ran a successful weapon only scan. 

The scanner crashes during the Artifact only scan. So I assume the issue was with artifacts all along.
The scanner has crashed in the same spot each time when running an artifact only scan. When the bottom left artifact counter is on 131/213. This should hit 196/213 when completed.

The scanning in game had finished by that point. The artifact menu had closed and the paimon menu had then reopens and there are no more inputs in to my Genshin Game. At this point the Inventory Scanner app was still checking through he artifacts. The section in the app that shows the screengrabs of each artefact was cycling through what it had seen.

Maybe a issue with my laptop being slow and not processing each artifact fast enough

I have gone back to version 1.0 and used that, it worked perfectly
